### ISSUE TITLE
Packets coalesced in wrong order

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#294]
Looking at the log of a session from client to server, I see that the second client packet includes four coalesced packets, which is good, but the packets are not in order of sequence, which is questionable:
```
7c4dd96ccd38bc4: Sending packet type: 5 (0rtt protected), S0, Version ff00000e,
7c4dd96ccd38bc4:     <b95bfc6b7bb0cd1e>, <bb2543195761ecb0>, Seq: 1, pl: 13
7c4dd96ccd38bc4:     Prepared 13 bytes
7c4dd96ccd38bc4:     ACK (nb=0), 0-1
7c4dd96ccd38bc4:     Crypto HS frame, offset 0, length 4: 05000000

7c4dd96ccd38bc4: Sending packet type: 2 (initial), S0, Version ff00000e,
7c4dd96ccd38bc4:     <b95bfc6b7bb0cd1e>, <bb2543195761ecb0>, Seq: 1, pl: 6
7c4dd96ccd38bc4:     Prepared 6 bytes
7c4dd96ccd38bc4:     ACK (nb=0), 0

7c4dd96ccd38bc4: Sending packet type: 4 (handshake), S0, Version ff00000e,
7c4dd96ccd38bc4:     <b95bfc6b7bb0cd1e>, <bb2543195761ecb0>, Seq: 0, pl: 61
7c4dd96ccd38bc4:     Prepared 61 bytes
7c4dd96ccd38bc4:     ACK (nb=0), 0
7c4dd96ccd38bc4:     Crypto HS frame, offset 0, length 52: 14000030d5584782...

7c4dd96ccd38bc4: Sending packet type: 6 (1rtt protected phi0), S1,
7c4dd96ccd38bc4:     <b95bfc6b7bb0cd1e>, Seq: 2 (2)
7c4dd96ccd38bc4:     Prepared 115 bytes
7c4dd96ccd38bc4:     ACK (nb=0), 0-1
7c4dd96ccd38bc4:     Stream 4, offset 0, length 16, fin = 1: 474554202f746573...
7c4dd96ccd38bc4:     Stream 8, offset 0, length 22, fin = 1: 474554202f646f63...
7c4dd96ccd38bc4:     Stream 12, offset 0, length 15, fin = 1: 474554202f6d6169...
7c4dd96ccd38bc4:     Stream 16, offset 0, length 24, fin = 1: 474554202f776172...
7c4dd96ccd38bc4:     Stream 20, offset 0, length 17, fin = 1: 474554202f656e2f...

7c4dd96ccd38bc4: Sending 357 bytes to 34.238.92.221:4433 at T=0.166641 (57428de34bf19)
```

**Comment by huitema** [comment]
Also, it appears that the 0-RTT packet carries an ACK for the 1-RTT packets that were just received. That's probably wrong too.

**Comment by huitema** [comment]
But having 4 different encryption levels in a single packet is awesome!

**Comment by huitema** [comment]
Fixed in PR #296
